src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -172,6 +172,9 @@
     @Transient
     private Long studentAssignedTeamIdTransient; // id of the team that the logged-in user is assigned to (only relevant if team mode is enabled)
 
+    @Transient
+    private String testRepositoryUrlTransient;
+
```

Comment:
You misunderstood me. Do not set the string again here. Use a transient Boolean in ProgrammingExercise called isLocalSimulation (or similar) and only set this to true if the url includes artemislocal (Same Check as on the client)

Reply:
please still move this to `ProgrammingExercise.java`

Reply:
Alright, i updated the transient property